Runtime/Tracking/Collision/CollisionIgnorer.cs

```diff
@@ -0,0 +1,213 @@
+        /// </summary>
+        /// <param name="source">The source to ignore collisions from.</param>
+        protected virtual void AddSource(GameObject source)
+        {
+            ToggleCollisions(source, Sources, Targets, true);
+        }
+
+        /// <summary>
+        /// Removes an existing source from ignoring collisions with all of the <see cref="Targets"/> colliders.
+        /// </summary>
+        /// <param name="source">The source to restore collisions with.</param>
+        protected virtual void RemoveSource(GameObject source)
+        {
+            ToggleCollisions(source, Sources, Targets, false);
+        }
+
+        /// <summary>
+        /// Adds a new target to ignore colliding on from of the <see cref="sources"/> colliders.
+        /// </summary>
+        /// <param name="target">The target to ignore collisions on.</param>
+        protected virtual void AddTarget(GameObject target)
+        {
+            ToggleCollisions(target, Targets, Sources, true);
+        }
+
+        /// <summary>
+        /// Removes an existing target from having collisions ignored with all of the <see cref="sources"/> colliders.
+        /// </summary>
+        /// <param name="target">The target to restore collisions on.</param>
+        public virtual void RemoveTarget(GameObject target)
+        {
+            ToggleCollisions(target, Targets, Sources, false);
+        }
+
+        /// <summary>
+        /// Sets the collision state between <see cref="Sources"/> and <see cref="Targets"/>.
+        /// </summary>
+        /// <param name="state">Whether to ignore collisions or not.</param>
+        protected virtual void ToggleCollisions(bool state)
+        {
+            foreach (GameObject source in Sources.SubscribableElements)
+            {
+                ToggleCollisions(source, Sources, Targets, state);
+            }
+        }
+
+        /// <summary>
+        /// Sets the collision state between the source and targets.
+        /// </summary>
+        /// <param name="source">The source to set the collision state on.</param>
+        /// <param name="targets">A collection of targets to set the collision state on.</param>
+        /// <param name="state">Whether to ignore collisions or not.</param>
+        protected virtual void ToggleCollisions(GameObject source, GameObjectObservableList sources, GameObjectObservableList targets, bool state)
+        {
+            if (source == null || (!state && sources.Contains(source)))
+            {
+                return;
+            }
+
+            List<Collider> sourceColliders = new List<Collider>();
+            List<Collider> targetColliders = new List<Collider>();
+
+            sourceColliders.AddRange(source.GetComponentsInChildren<Collider>());
+
+            foreach (GameObject target in targets.SubscribableElements)
+            {
+                if (target != null)
```

Comment:
Didn't we say we want `null` values to be treated as errors and can thus ignore it here as the null ref exception being thrown without the if is enough for the user to know?

Reply:
Not in this case, because it's called OnDisable and the target can be destroyed when closing and if this runs still and the target is null (because it's been destroyed) then it will throw an error

Reply:
👍